src/Microsoft.Health.Dicom.Web/Program.cs

```diff
@@ -23,9 +23,17 @@ public static void Main(string[] args)
                     IConfigurationRoot builtConfig = builder.Build();
 
                     var keyVaultEndpoint = builtConfig["KeyVault:Endpoint"];
+                    var userAssignedAppId = builtConfig["DicomServer:ServerIdentity:UserAssignedAppId"];
+                    string tokenProviderConnectionString = null;
+
+                    if (!string.IsNullOrEmpty(userAssignedAppId))
+                    {
+                        tokenProviderConnectionString = $"RunAs=App;AppId={userAssignedAppId}";
+                    }
+
                     if (!string.IsNullOrEmpty(keyVaultEndpoint))
                     {
-                        var azureServiceTokenProvider = new AzureServiceTokenProvider();
+                        var azureServiceTokenProvider = new AzureServiceTokenProvider(connectionString: tokenProviderConnectionString);
```

Comment:
This looks to only change the usage of the keyvault provider if it's being used. It doesn't seem like it would have an affect on the [SQL Provider](https://github.com/microsoft/healthcare-shared-components/blob/57e8f06dde0e27dfccbc9281e78665492d6c203f/src/Microsoft.Health.SqlServer/Registration/SqlServerBaseRegistrationExtensions.cs#L84) or the [Storage Provider](https://github.com/microsoft/healthcare-shared-components/blob/ab91e7a841b23b10ea5a611715528892d59fcdc1/src/Microsoft.Health.Blob/Features/Storage/BlobClientFactory.cs#L27)

Reply:
Resolved by https://github.com/microsoft/healthcare-shared-components/pull/272